pkg/controller/controller_instance.go

```diff
@@ -580,19 +642,14 @@ func findServicePlan(name string, plans []v1alpha1.ServicePlan) *v1alpha1.Servic
 	return nil
 }
 
-// updateInstanceCondition updates the given condition for the given Instance
-// with the given status, reason, and message.
-func (c *controller) updateInstanceCondition(
-	instance *v1alpha1.Instance,
+// setInstanceCondition sets a single condition on an Instance's status.  The
+// instance parameter should always be a deep copy of the object being updated
+// to avoid mutating caches.
+func setInstanceCondition(instance *v1alpha1.Instance,
```

Comment:
What's the end state of the condition here? 
The instance has only exactly one condition set and no others?
If it had multiple conditions set coming in it ends up with only exactly one going out?